Sources/Segment/Utilities/JSON.swift

```diff
@@ -267,9 +267,73 @@ extension JSON {
     }
 }
 
+// MARK: - Mapping
+
+extension JSON {
+    /// Maps keys supplied, in the format of ["Old": "New"].  Gives an optional value transformer that can be used to transform values based on the final key name.
+    /// - Parameters:
+    ///   - keys: A dictionary containing key mappings, in the format of ["Old": "New"].
+    ///   - valueTransform: An optional value transform closure.  Key represents the new key name.
+    public func mapKeys(_ keys: [String: String], valueTransform: ((_ key: String, _ value: Any) -> Any)? = nil) throws -> JSON {
+        guard let dict = self.dictionaryValue else { return self }
+        let mapped = try dict.mapKeys(keys, valueTransform: valueTransform)
+        let result = try JSON(mapped)
+        return result
+    }
+}
 
 // MARK: - Helpers
 
+extension Dictionary where Key == String, Value == Any {
+    internal func mapKeys(_ keys: [String: String], valueTransform: ((_ key: Key, _ value: Value) -> Any)? = nil) throws -> [Key: Value] {
```

Comment:
Do you need `= nil` on there?

Reply:
Yeah, cuz I want it to be optional.  They likely just want to map keys, at which point, the values won't be transformed.